include/FileDialog.h

```diff
@@ -37,20 +36,47 @@ class LMMS_EXPORT FileDialog : public QFileDialog
 {
 	Q_OBJECT
 public:
+	enum Operation {
+		OpBegin = 0, // This variant is purely for looping conveviences.
+		OpGeneric = OpBegin,
+		OpProject,
+		OpMidi,
+		OpPreset,
+		OpPlugin,
+		OpSample,
+		OpSong,
+		OpEnd, // This variant is purely for looping conveviences
+	};
```

Comment:
I would recommend using an [`enum class`](https://en.cppreference.com/w/cpp/language/enum#Scoped_enumerations) here instead, and updating the code that uses `Operation` accordingly.